test/Rollup.spec.ts

```diff
@@ -319,26 +414,993 @@ contract("Rollup", async function (accounts) {
+    var AliceAccountMP = {
+      accountIP: {
+        pathToAccount: AliceAccountPath,
+        account: {
+          ID: Alice.AccID,
+          tokenType: Alice.TokenType,
+          balance: Alice.Amount,
+          nonce: Alice.nonce,
+        },
+      },
+      siblings: AliceAccountSiblings,
+    };
+
+    Alice.Amount -= Number(tx.amount);
+    Alice.nonce++;
+
+    var UpdatedAliceAccountLeaf = await createLeaf(Alice);
+
+    // bob balance tree merkle proof
+    var BobAccountSiblings: Array<string> = [
+      UpdatedAliceAccountLeaf,
+      utils.getParentLeaf(coordinator_leaves[0], coordinator_leaves[1]),
+      zeroHashes[2],
+      zeroHashes[3],
+    ];
+    var leaf = BobAccountLeaf;
+    var BobAccountPath: string = "3";
+    var isBobValid = await MTutilsInstance.verifyLeaf(
+      currentRoot,
+      leaf,
+      BobAccountPath,
+      BobAccountSiblings
+    );
+
+    var BobAccountMP = {
+      accountIP: {
+        pathToAccount: BobAccountPath,
+        account: {
+          ID: Bob.AccID,
+          tokenType: Bob.TokenType,
+          balance: Bob.Amount,
+          nonce: Bob.nonce,
+        },
+      },
+      siblings: BobAccountSiblings,
+    };
+
+    Bob.Amount += Number(tx.amount);
+    var accountProofs = {
+      from: AliceAccountMP,
+      to: BobAccountMP,
+    };
+
+    // process transaction validity with process tx
+    var result = await rollupCoreInstance.processTx(
+      currentRoot,
+      accountRoot,
+      tx,
+      alicePDAProof,
+      accountProofs
+    );
+
+    var falseResult = await falseProcessTx(
+      fraudProofInstance,
+      currentRoot,
+      accountRoot,
+      tx,
+      alicePDAProof,
+      accountProofs
+    );
+    console.log("result from falseProcessTx: " + falseResult);
+    console.log("result from processTx: " + JSON.stringify(result));
+    
+    assert.equal(result[3], ERR_TOKEN_AMT_INVAILD, "false Error Id. It should be `2`.");
+
+    await compressAndSubmitBatch(tx, falseResult)
+
+    falseBatchTwo = {
+      batchId: 0,
+      txs: [tx],
+      batchProofs: {
+        accountProofs: [accountProofs],
+        pdaProof: [alicePDAProof]
+      }
+    }
+    
+    let batchId = await rollupCoreInstance.numOfBatchesSubmitted();
+    falseBatchTwo.batchId = Number(batchId) - 1;
+  });
+
+  it("dispute batch false 3rd batch(Tx amount 0)", async function () {
+    await rollupCoreInstance.disputeBatch(
+      falseBatchTwo.batchId,
+      falseBatchTwo.txs,
+      falseBatchTwo.batchProofs,
+    );
+    
+    
+    let batchId = await rollupCoreInstance.numOfBatchesSubmitted();
+    let batchMarker = await rollupCoreInstance.invalidBatchMarker();
+    assert.equal(batchMarker, "0", "batchMarker is not zero");
+    assert.equal(batchId -1 , falseBatchTwo.batchId - 1, "batchId doesnt match");
+    Alice.Amount += falseBatchTwo.txs[0].amount;
+    Bob.Amount -= falseBatchTwo.txs[0].amount;
+    Alice.nonce--;
+  })
+
+  // it("submit new batch 4nd", async function () {
```

Comment:
@thrilok209 can you please remove the commented part if it's not needed, the test file is very long as is 😰